projects/plugins/boost/app/features/optimizations/cloud-css/Cloud_CSS.php

```diff
@@ -37,6 +38,10 @@ public function setup() {
 		REST_API::register( $this->get_endpoints() );
 		Critical_CSS_Invalidator::init();
 
+		// Priority must be greater than 10 to run after invalidator.
+		add_action( 'handle_theme_change', array( $this, 'generate_cloud_css' ), 11, 0 );
```

Comment:
I think we should avoid avoid timing dependencies like this, where one action handler depends on the outcome of another with a lower priority, because it's fragile and prone to break if we ever change the priority of the other `handle_theme_change` handler.

IMHO we should add a new action which is called after the Critical CSS is invalidated - such as `'critical_css_invalidated'`, and listen for _that_ here instead.

That removes a fragile priority-based order-dependency, _and_ means that if we add more reasons to invalidate the Critical CSS, it will automatically be picked up here.

Reply:
🤔 Hmm. That is a fair point.

Does the Critical CSS Invalidator know whether it's invalidating due to uninstallation, or if it's invalidating due to a settings change? Would it make sense to make it only send out an action like `critical_css_invalidated` if it's not uninstalling?